sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -1977,6 +1977,26 @@ def set_premium_page_blob_tier(self, premium_page_blob_tier, **kwargs):
         except StorageErrorException as error:
             process_storage_error(error)
 
+    def _parse_snapshot_value(self, snapshot):
+        # type (Union[str, Dict[str, Any], BlobClient, BlobProperties]) -> Dict[str,str]
+        if not snapshot:
+            return {}
+        try:
+            result = urlparse(snapshot)
+            if all([result.scheme, result.netloc, result.path]):
+                return {'prev_snapshot_url': snapshot}
```

Comment:
it seems we haven't decided to merge managed disk diff into this method yet?

Reply:
This is a draft PR until the discussion has concluded.